### PR TITLE
Offline make request

### PR DESCRIPTION
**Pull request by kieter**
You can now make rider requests while offline and have them be added to elastic search once connectivity comes back. Covers [#146](https://github.com/CMPUT301F16T01/Carrier/issues/146), [#202](https://github.com/CMPUT301F16T01/Carrier/issues/202), [#211](https://github.com/CMPUT301F16T01/Carrier/issues/211)

See it in action:
Log in online, go offline, make a request, go back online, swipe down to refresh.
If you close the app and open the app now you should see that the requests added offline persist online (and that the route is calculated for those requests once back online).

### CODE REVIEW

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java [hunk 1]
```diff
@@ -44,6 +52,9 @@
     /** The file name of the locally saved offered driver requests. */
     private static final String DRIVER_FILENAME = "DriverRequests.sav";
 
+    /** The file name of the locally saved offline rider requests. */
+    private static final String OFFLINE_FILENAME = "OfflineRequests.sav";
```
Comment (sehoven) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java]: Could you call this OFFLINE_REQUEST_FILENAME or something more descriptive to show that it is dealing with the requests specifically?

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java [hunk 2]
```diff
@@ -87,11 +98,19 @@ public static String addRequest(Request request) {
...
+            boolean internetConnection = ConnectionChecker.isThereInternet();
             // If there is internet we update ElasticSearch with the new request.
-            if (ConnectionChecker.isThereInternet()) {
+            if (internetConnection) {
                 ElasticRequestController.AddRequestTask art = new ElasticRequestController.AddRequestTask();
                 art.execute(request);
             }
+
+            // If there is no internet, add to the offline request queue and save it
+            if (!internetConnection) {
```
Comment (sehoven) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java]: Maybe just make this an if-else?

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java [hunk 3]
```diff
@@ -87,11 +98,19 @@ public static String addRequest(Request request) {
...
+                offlineRiderRequests.add(request);
+                Toast.makeText(saveContext, "Offline saving" + Integer.toString(offlineRiderRequests.size()), Toast.LENGTH_SHORT).show();
```
Comment (sehoven) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java]: Delete this Toast.

**Review by sehoven — CHANGES_REQUESTED** [review]
If I make 2 requests while offline, then I go back online and refresh, only one request shows up. If I refresh again, both show up (as I expect).

Try putting it in onPostExecute

**Comment by sehoven** [comment]
I was trying to run a branch that was branched off of your branch to work on a different issue and it won't let me do anything on the app. It crashes every time I try to do pretty much anything (and I'm not sure when this started happening) .

screen shot 2016-11-27 at 3 24 46 am

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java (hunk 1 above)
Comment (kieter) [reply app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java]: Did itttt

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java [hunk 4]
```diff
@@ -347,22 +369,81 @@ public static RequestList fetchAllRequestsWhereRider( User rider ) {
...
+     * Get address string from a geo point
+     *
+     * @return the address as a string.
+     */
+    // see code attribution
+    public static String getAddress(Context context, double latitude, double longitude) {
+        String pointAddress;
+        try {
+            Geocoder geocoder = new Geocoder(context);
+            List<Address> addresses = geocoder.getFromLocation(latitude, longitude, 1);
```
Comment (sehoven) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java:440]: Import List type (from me merging my branch)

**Review by sehoven — APPROVED** [review]

**Pull request merged**